docker/README.md

```diff
@@ -0,0 +1,84 @@
+### Running Naksha in container
+
+Naksha can be run in container as well. So far, only locally build image can be used.\
+To get Naksha container running, one must do the following:
+
+1) CD into the root project directory (we need this because build context needs to access files that
+   don't belong to `docker` directory).
+     ```shell
+    cd ..
+    ```
+2) Build the fat jar:
+     ```shell
+    ./gradlew shadowJar
+    ```
+3) Build the local image
+    ```shell
+   docker build -t local-naksha-app -f docker/Dockerfile .
+    ```
+4) Run the container:
+   There are two optional environment variables that one can specify when running Naksha conrtainer
+    - `NAKSHA_CONFIG_ID`: id of naksha configaration to use, `test-config` by default
+    - `NAKSHA_ADMIN_DB_URL`: url of database for Naksha app to
+      use, `jdbc:postgresql://localhost:5432/postgres?user=postgres&password=password&schema=naksha&app=naksha_local&id=naksha_admin_db`
+      by default
+
+   When connecting Naksha app to database, one has to consider container networking - if your
+   database is running locally, then you need to instruct the container to utilize host's
+   network (`--network=host`).\
+   Putting it all together the typical command you would use is:
+   ```shell
+   docker run \            
```

Comment:
I suggest validating once that we are able to load the Swagger and access the service from local machine using Http port ... if it requires specifying a port while running the command, then pls update this command.
If it works by default, then this comment can be ignored.

Reply:
![image](https://github.com/xeus2001/xyz-hub/assets/9114956/238e30eb-9952-4e42-b9ce-f2b60951a5d3)

When testing this I stumbled upon one issue with my previous setup. I explain the problem and solution below - it was a nice problem to solve ;) To sum it up quickly: I ditched `--network=host` and started to use `host.docker.internal` instead of `localhost`/`127.0.0.1`.

Initially, I was using `--network=host` which makes docker use the `host` network driver. This exposes all of the container ports to the host and also allows the container to use all ports opened by the localhost - at least in theory ;) 
In practice, this "two-way deal" works only on Linux. On Mac & Windows, Docker/Podman respects `--network=host` only partially - the container can use all of the localhosts network resources, but publishing ports is supported only on Linux.
That being said, the combination of `--network=host` and `-p 8080:8080` also won't work - it is by design. The reason for this design is explained in [Docker's docs for Host driver](https://docs.docker.com/network/drivers/host/):
```
Given that the container does not have its own IP-address when using host mode networking, port-mapping doesn't take effect, and the -p, --publish, -P, and --publish-all option are ignored
```

So given that `--host=networking` is not fully supported on Mac/Windows & `-p 8080:8080` does not work with `--host-networking`, I realized we can't use `host` driver. 
Still, we needed a way to:
- connect our container to some local database
- publish 8080

Publishing is straightforward (`-p 8080:8080` will work with the default `bridge` network driver) but accessing local resources was something I had to explore a bit more.
I found out a nice [workaround developed by Docker team: special DNS entry for Mac and Windows](https://docs.docker.com/desktop/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host)

```
(...)  We recommend that you connect to the special DNS name host.docker.internal, which resolves to the internal IP address used by the host.
```

So basically, whenever we have a resource that is served by our local Mac/Windows machine, we need to refer to `host.docker.internal`

We need to remember about this when we'll go into CI/CD where hosts have Linux - we might face the problem of `host.docker.internal` not being resolved. Then we would simply have to do one of the following:
- declare Docker compose with services & network, where we connect containerized Postgres with containerized Naksha (I think this is the way to go anyway)
- simply override the `NAKSHA_ADMIN_DB_URL` env variable and use good old `localhost` / `127.0.0.1`

Reply:
The reason why I did not specify `-p 8080:8080` in the `run` command is based on the assumption that `--network=host` would work as I expected it (I have used this driver on Linux machines before). My mistake was that I only crosschecked Naksha-Postgres connection without having direct interaction with Naksha in the first place.

Reply:
* supporting arm64v8 and amd64 (even by supplying argument should be sufficient)
* update command by removing all extra whitespaces
* validate if localhost required for Podman only (if not, we can remove that from container start command)
* by default we can point NAKSHA_ADMIN_DB_URL to `host.docker.internal` in Dockerfile
* explicitly specify that the same swagger endpoint should work (http://localhost:8080/hub/swagger/index.html)

Reply:
all done